packages/react-reconciler/src/ReactFiberLane.js

```diff
@@ -7,16 +7,682 @@
+    let lane = findLane(
+      TransitionShortUpdateRangeStart,
+      TransitionShortUpdateRangeEnd,
+      wipLanes | pendingLanes,
+    );
+    if (lane === NoLane) {
+      lane = findLane(
+        TransitionShortUpdateRangeStart,
+        TransitionShortUpdateRangeEnd,
+        wipLanes,
+      );
+      if (lane === NoLane) {
+        lane = TransitionShortHydrationLane;
+      }
+    }
+    return lane;
+  }
+  if (lanePriority === TransitionLongLanePriority) {
+    let lane = findLane(
+      TransitionLongUpdateRangeStart,
+      TransitionLongUpdateRangeEnd,
+      wipLanes | pendingLanes,
+    );
+    if (lane === NoLane) {
+      lane = findLane(
+        TransitionLongUpdateRangeStart,
+        TransitionLongUpdateRangeEnd,
+        wipLanes,
+      );
+      if (lane === NoLane) {
+        lane = TransitionLongHydrationLane;
+      }
+    }
+    return lane;
+  }
+  invariant(
+    false,
+    'Invalid transition priority: %s. This is a bug in React.',
+    lanePriority,
+  );
+}
+
+function findLane(start, end, skipLanes) {
+  // This finds the first bit between the `start` and `end` positions that isn't
+  // in `skipLanes`.
+  // TODO: This will always favor the rightmost bits. That's usually fine
+  // because any bit that's pending will be part of `skipLanes`, so we'll do our
+  // best to avoid accidental entanglement. However, lanes that are pending
+  // inside an Offscreen tree aren't considered "pending" at the root level. So
+  // they aren't included in `skipLanes`. So we should try not to favor any
+  // particular part of the range, perhaps by incrementing an offset for each
+  // distinct event. Must be the same within a single event, though.
+  const bitsInRange = ((1 << (end - start)) - 1) << start;
+  const possibleBits = bitsInRange & ~skipLanes;
+  const leastSignificantBit = possibleBits & -possibleBits;
+  return leastSignificantBit;
+}
+
+function getLowestPriorityLane(lanes: Lanes): Lane {
+  // This finds the most significant non-zero bit.
+  const index = 31 - clz32(lanes);
+  return index < 0 ? NoLanes : 1 << index;
+}
+
+export function pickArbitraryLane(lanes: Lanes): Lane {
+  return getLowestPriorityLane(lanes);
+}
+
+export function includesSomeLane(a: Lanes | Lane, b: Lanes | Lane) {
+  return (a & b) !== NoLanes;
+}
+
+export function isSubsetOfLanes(set: Lanes, subset: Lanes | Lane) {
+  return (set & subset) === subset;
+}
+
+export function mergeLanes(a: Lanes | Lane, b: Lanes | Lane): Lanes {
+  return a | b;
+}
+
+export function removeLanes(set: Lanes, subset: Lanes | Lane): Lanes {
+  return set & ~subset;
+}
+
+// Seems redundant, but it changes the type from a single lane (used for
+// updates) to a group of lanes (used for flushing work).
+export function laneToLanes(lane: Lane): Lanes {
+  return lane;
+}
+
+export function higherPriorityLane(a: Lane, b: Lane) {
+  // This works because the bit ranges decrease in priority as you go left.
+  return a !== NoLane && a < b ? a : b;
```

Comment:
Is it possible that `b` equals to `NoLane`?

Reply:
In practice no, it's always `SelectiveHydrationLane` but should probably rename/refactor this to something more specific so it's not used elsewhere.

Reply:
Good catch :)